pages/profile/[name]/[[...tab]]/index.tsx

```diff
@@ -172,6 +179,71 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
     profilePageProps.totalRows = query.totalRows;
   }
 
+  if (profileTab === ProfileTab.Profile) {
+    const levelsCompletedByDifficultyData = await StatModel.aggregate([
+      {
+        $match: {
+          $and: [ {userId: user._id }, { complete: true}],
+        },
+      },
+      {
+        $lookup: {
+          from: 'levels',
+          localField: 'levelId',
+          foreignField: '_id',
+          as: 'levelInfo',
+        },
+      },
+      {
+        $unwind:"$levelInfo"
+      },
+      {
+        $project: {
+          difficulty: "$levelInfo.calc_difficulty_estimate"
+        }
+      }
+    ]);
+
+    let levelsCompletedByDifficulty:levelsCompletedByDifficultyInterface = {};
+    levelsCompletedByDifficulty["Pending"] = 0;
+    const difficultyList = getDifficultyList();
+
+    for (let i = 0; i < difficultyList.length; i++) {
+      levelsCompletedByDifficulty[difficultyList[i].name] = 0;
+    }
+
+    for (let i = 0; i < levelsCompletedByDifficultyData.length; i++) {
+      const difficultyLookup = getDifficultyFromValue(levelsCompletedByDifficultyData[i].difficulty)
+      
+      levelsCompletedByDifficulty[difficultyLookup.name] = levelsCompletedByDifficulty[difficultyLookup.name] + 1;
+    }
+
+    const numWidth = 50;
+    var levelCompletedByRankTable = [
+      <tr key={'statistics-header'} style={{ backgroundColor: 'var(--bg-color-2)' }}>
```

Comment:
In general JSX shouldn't be returned from `getServerSideProps`. I think the best option overall for the structure here is to return just an array of numbers, which should be the minimal amount of data needed from the server in order to render what we want.

Reply:
So the processing to the covert the data into a table should happen in the profile page props?

Reply:
Actually maybe it would be better to convert the data into a map of difficulty to completed count on the server side. But the code to generate the table itself should happen within `ProfilePage`,